uint8 from_vertex
uint8 to_vertex
float32 deposit
